haystack_experimental/components/agents/human_in_the_loop/types.py

```diff
@@ -62,8 +62,13 @@ def from_dict(cls, data: dict[str, Any]) -> "ConfirmationPolicy":
 
 
 class ConfirmationStrategy(Protocol):
-    def run(
-        self, tool_name: str, tool_description: str, tool_params: dict[str, Any], tool_call_id: Optional[str] = None
+    def run(  # pylint: disable=R0917
+        self,
+        tool_name: str,
+        tool_description: str,
+        tool_params: dict[str, Any],
+        tool_call_id: Optional[str] = None,
+        run_context: Optional[dict[str, Any]] = None,
```

Comment:
See https://github.com/deepset-ai/haystack-experimental/pull/401#discussion_r2567842451.
Do we really need `run_context` in the Protocol? (Maybe yes but let's clarify this point)

Reply:
If anything I think we just need to allow generic kwargs. See my comment here https://github.com/deepset-ai/haystack-experimental/pull/401#discussion_r2580269597 (I think this is what you were already getting at in your [comment](https://github.com/deepset-ai/haystack-experimental/pull/401#discussion_r2567842451))